engine/src/main/java/org/terasology/engine/modes/StateLoading.java

```diff
@@ -88,6 +90,10 @@
     private int progress;
     private int maxProgress;
 
+    private static final long chunkGenerationTimeout = 20000;
```

Comment:
All caps on constants please, and toss it up to the top of the class :-) 

Default seems OK. Maybe there'd be a way to make it part of `config.cfg` or even expose it in the settings in case any players need a longer timeout?